src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEFieldSymbol.cs

```diff
@@ -20,6 +20,59 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE
     /// </summary>
     internal sealed class PEFieldSymbol : FieldSymbol
     {
+        private struct PackedFlags
+        {
+            // Layout:
+            // |..............................|vvvvv|
+            //
+            // f = FlowAnalysisAnnotations. 5 bits (4 value bits + 1 completion bit).
+
+            private const int HasDisallowNullAttribute = 0x1 << 0;
+            private const int HasAllowNullAttribute = 0x1 << 1;
+            private const int HasMaybeNullAttribute = 0x1 << 2;
+            private const int HasNotNullAttribute = 0x1 << 3;
```

Comment:
Handling individual bits here is not necessary since `FlowAnalysisAnnotations` is already stored as a set of bits (see `PEParameterSymbol.PackedFlags`).

Reply:
PEParameterSymbol can store all the possible annotations, but here we only need to store only four, which are not even contiguous (see below). So I think we have to deal with individual bits/flags.
```C#
    [Flags]
    internal enum FlowAnalysisAnnotations
    {
        None = 0,
        AllowNull = 1 << 0,
        DisallowNull = 1 << 1,
        MaybeNullWhenTrue = 1 << 2,
        MaybeNullWhenFalse = 1 << 3,
        MaybeNull = MaybeNullWhenTrue | MaybeNullWhenFalse,
        NotNullWhenTrue = 1 << 4,
        NotNullWhenFalse = 1 << 5,
        NotNull = NotNullWhenTrue | NotNullWhenFalse,
        AssertsTrue = 1 << 6,
        AssertsFalse = 1 << 7,
    }
```

---
In reply to: [296767023](https://github.com/dotnet/roslyn/pull/36704#discussion_r296767023) [](ancestors = 296767023)